### ISSUE TITLE
Trying to access a file uploader inside an iframe causes RangeError: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by srajagopalan1712** [angular/protractor#1956]
I am trying to upload a file and the file selector in side an iframe. Belwo is the html and the protractor code to upload the file. When I run the test I get this error

  Message:
     RangeError: Maximum call stack size exceeded
   Stacktrace:
     RangeError: Maximum call stack 

Any help would be appreciated!! Thx!

var fileToUpload = '../myPath/Sample.txt';
var absolutePath = path.resolve(__dirname, fileToUpload);
browser.switchTo().frame(element(by.id('uploader'))); 

//<iframe frameborder="0" id="uploader" onload="onLoadIFrame();" src=".path.."></iframe> 
//<input accept="" accesskey="" alt="" class="" dir="" id="..." name="..." type="file">

$('input[type="file"]').sendKeys(absolutePath);
browser.sleep(3000);
element(by.linkText('Upload Files')).click(); // click button to upload
browser.switchTo().defaultContent(); // switch context back to main page

Also the parent page is a non-angular page and I have set
browser.ignoreSynchronization = true; 

I am not sure if I should set it again when I get into the iframe or when I get back out.

**Comment by srajagopalan1712** [comment]
Seems to be the same issue as https://github.com/angular/protractor/issues/1846. I tried the workaround mentioned there. It seems the frame is selected and the file element is now visible to the test. While the test runs, in the browser I see the frame area looks liek a error 404 page and I do not see uploading happen in the backend too. So not if the lines to snedKeys (path), click Upload happened and content switch back to main browser happened. I have a Link in main page to click for my test to complete the wizard. Thats not happening, so content is not switched back too as far as I see

**Comment by juliemr** [comment]
Are you using the latest version of Protractor? This should not happen with 2.0.0.

**Comment by srajagopalan1712** [comment]
I updated to 2.0.0 and I realized my grunt-protractor-runner would not run as it needs the older protractor. i removed all references to it and am directly running protractor now. I upgraded the chromedriver to latest. Now I realize that even the basic button click on my Step 1 of my wizard will not click. My file uploader was in Step 5 and itw as working well until that pointin older Protractor version. All I am doing to go to Step 2 is this:
element(by.css('.form-group > .btn-primary')).click();

For some reason it is looking fpr element in teh next page without moving out of this and fails saying that element is not found. Unless this click happens, it will not find it. Note I mentioned above that my UI is non-angular and I have browser.ignoreSynchronization = true;  on my beforeEach.

Has there been any change in the new version that should break this, I am not clear.

**Comment by juliemr** [comment]
You can see the full changelog at https://github.com/angular/protractor/blob/master/CHANGELOG.md, but from your quick description I don't see anything that touches a known breaking change.

**Comment by srajagopalan1712** [comment]
That was a quick response. Y I read the changes. How can a basic button click taht worked earlier and not now is where I am lost.

**Comment by srajagopalan1712** [comment]
Will my fiile uploader UI in an iframe not work in an earlier version of protractor? 1.8? I thought since my frame is inside my non-angular page, the fix would be something related to ignoreSynchronization inside the iframe and making sure I come back to main page right. I am not sure moving to Protractor 2 is working right for the rest of my test.

**Comment by srajagopalan1712** [comment]
Protractor 2.0 made this worse for me. I am on macbook air and it went on node out of memory error and such. I moved back to 1.7 and since my main page and iframe is non-angular, I had to do         browser.ignoreSynchronization = true; inside the frame after I move into it as well. Also after I move out of frame to main page, I had to sleep extra. 

While inside the frame, I had to access my file element and upload button using 
browser.driver.findElement(by.css('input[type="file"]')).sendKeys(absolutePath); 
$('input[type="file"]').sendKeys(absolutePath); is not right.
